### PR TITLE
Upgrade dependencies for @angular, @okta

### PR DESCRIPTION
**Pull request by cavazosjoe**
@angular to v18.2.7
@okta/okta-angular to v6.4.0
@okta/okta-auth-js to v7.8.0

### CODE REVIEW

File: src/app/app.config.ts [hunk 1]
```diff
@@ -1,9 +1,12 @@
 const { CLIENT_ID, ISSUER, OKTA_TESTING_DISABLEHTTPSCHECK } = process.env;
 
+let clientId:string = CLIENT_ID!
```
Comment (alisaduncan) [inline src/app/app.config.ts:3]:
```suggestion
```

File: src/app/app.config.ts [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
+let issuer:string = ISSUER!
```
Comment (alisaduncan) [inline src/app/app.config.ts:4]:
```suggestion
```

File: src/app/app.config.ts [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
+
 export default {
   oidc: {
-    clientId: CLIENT_ID,
-    issuer: ISSUER,
+    clientId: clientId,
```
Comment (alisaduncan) [inline src/app/app.config.ts:8]:
```suggestion
    clientId: CLIENT_ID as string,
```

File: src/app/app.config.ts [hunk 4]
```diff
@@ -1,9 +1,12 @@
...
+    issuer: issuer,
```
Comment (alisaduncan) [inline src/app/app.config.ts:9]:
```suggestion
    issuer: ISSUER as string,
```

File: src/app/app.module.ts [hunk 5]
```diff
@@ -10,21 +10,21 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { BrowserModule } from '@angular/platform-browser';
-import { NgModule } from '@angular/core';
-import { HttpClientModule } from '@angular/common/http';
-import { Routes, RouterModule } from '@angular/router';
+import {BrowserModule} from '@angular/platform-browser';
+import {NgModule} from '@angular/core';
+import {provideHttpClient, withInterceptorsFromDi} from '@angular/common/http';
```
Comment (alisaduncan) [inline src/app/app.module.ts:15]: We can still use the `HttpClientModule` still. Let's revert this for now.

File: tsconfig.json [hunk 6]
```diff
@@ -5,14 +5,14 @@
     "baseUrl": "./",
     "outDir": "./dist/out-tsc",
     "forceConsistentCasingInFileNames": true,
+    "esModuleInterop": true,
```
Comment (alisaduncan) [inline tsconfig.json:8]: Please revert all changes in the tsconfig. Neither change is needed when i test

**Review by alisaduncan — COMMENTED** [review]
A few changes please

**Comment by alisaduncan** [comment]
Forgot to add, please regenerate package-lock file as a newer version of auth-js released last week that fixes a critical vuln. TY!

File: src/app/app.config.ts (hunk 1 above)
Comment (cavazosjoe) [reply src/app/app.config.ts]: Removed.

File: src/app/app.config.ts (hunk 2 above)
Comment (cavazosjoe) [reply src/app/app.config.ts]: Removed.

File: src/app/app.config.ts (hunk 3 above)
Comment (cavazosjoe) [reply src/app/app.config.ts]: Added this cast.

File: src/app/app.config.ts (hunk 4 above)
Comment (cavazosjoe) [reply src/app/app.config.ts]: Added this cast.

File: src/app/app.module.ts (hunk 5 above)
Comment (cavazosjoe) [reply src/app/app.module.ts]: Reverted this import.

File: tsconfig.json (hunk 6 above)
Comment (cavazosjoe) [reply tsconfig.json]: All changes reverted.

File: src/app/app.module.ts [hunk 7]
```diff
@@ -10,21 +10,21 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { BrowserModule } from '@angular/platform-browser';
-import { NgModule } from '@angular/core';
-import { HttpClientModule } from '@angular/common/http';
-import { Routes, RouterModule } from '@angular/router';
+import {BrowserModule} from '@angular/platform-browser';
```
Comment (alisaduncan) [inline src/app/app.module.ts:13]: Can we please revert the spacing and ordering changes in this file? this should be unneeded (and counters convention)

**Review by alisaduncan — CHANGES_REQUESTED** [review]
It looks like the build is failing. This is due to rollup dependency needing defined peerDeps in the package.json. I'll message you what needed to fix the build

**Comment by alisaduncan** [comment]
Closing this one in favor of #851

**Pull request closed without merging**